ocaml/xapi/xapi_xenops.ml

```diff
@@ -2049,6 +2049,7 @@ let update_vm ~__context id =
                       (fun (protocol, _) ->
                         let ref = Ref.make () in
                         let uuid = Uuid.to_string (Uuid.make_uuid ()) in
+                        let uri = Http.Url.maybe_wrap_IPv6_literal uri in
```

Comment:
Here, `uri` isn't just an address (see L2014), so you'll need to do this higher up.

Reply:
Indeed!